core/services/ccip/orm_test.go

```diff
@@ -0,0 +1,353 @@
+	t.Parallel()
+	ctx := testutils.Context(t)
+
+	orm, db := setupORM(t)
+
+	numJobs := 5
+	numAddresses := 10
+	numUpdatesPerAddress := 20
+	destSelector := uint64(1)
+
+	addrs := generateTokenAddresses(numAddresses)
+
+	updates := make(map[ccip.Address][]TokenPriceUpdate)
+	for _, addr := range addrs {
+		updates[addr] = generateTokenPriceUpdates(addr, numUpdatesPerAddress)
+	}
+
+	// 5 jobs, each inserting prices for 10 chains, with 20 updates per chain.
+	expectedPrices := make(map[ccip.Address]TokenPriceUpdate)
+	for i := 0; i < numJobs; i++ {
+		for addr, updatesPerAddr := range updates {
+			lastIndex := len(updatesPerAddr) - 1
+
+			err := orm.InsertTokenPricesForDestChain(ctx, destSelector, int32(i), updatesPerAddr[:lastIndex])
+			assert.NoError(t, err)
+			// make sure element at last index has unique timestamp
+			time.Sleep(1 * time.Millisecond)
+			err = orm.InsertTokenPricesForDestChain(ctx, destSelector, int32(i), updatesPerAddr[lastIndex:])
+			assert.NoError(t, err)
+
+			expectedPrices[addr] = updatesPerAddr[lastIndex]
+		}
+	}
+
+	// verify number of rows inserted
+	numRows := getTokenTableRowCount(t, db)
+	assert.Equal(t, numJobs*numAddresses*numUpdatesPerAddress, numRows)
+
+	prices, err := orm.GetTokenPricesByDestChain(ctx, destSelector)
+	assert.NoError(t, err)
+	// should return 1 price per source chain selector
+	assert.Equal(t, numAddresses, len(prices))
+
+	// verify getTokenPrices returns prices of latest timestamp
+	for _, price := range prices {
+		addr := price.TokenAddr
+		assert.Equal(t, expectedPrices[addr].TokenPrice, price.TokenPrice)
+	}
+
+	// after the initial inserts, insert new round of prices, 1 price per selector this time
+	var combinedUpdates []TokenPriceUpdate
+	for addr, updatesPerAddr := range updates {
+		combinedUpdates = append(combinedUpdates, updatesPerAddr[0])
+		expectedPrices[addr] = updatesPerAddr[0]
+	}
+
+	err = orm.InsertTokenPricesForDestChain(ctx, destSelector, 1, combinedUpdates)
+	assert.NoError(t, err)
+	assert.Equal(t, numJobs*numAddresses*numUpdatesPerAddress+numAddresses, getTokenTableRowCount(t, db))
+
+	prices, err = orm.GetTokenPricesByDestChain(ctx, destSelector)
+	assert.NoError(t, err)
+	assert.Equal(t, numAddresses, len(prices))
+
+	for _, price := range prices {
+		addr := price.TokenAddr
+		assert.Equal(t, expectedPrices[addr].TokenPrice, price.TokenPrice)
+	}
+}
+
+func TestORM_InsertAndDeleteTokenPrices(t *testing.T) {
+	t.Parallel()
+	ctx := testutils.Context(t)
+
+	orm, db := setupORM(t)
+
+	numAddresses := 10
+	numUpdatesPerAddress := 20
+	destSelector := uint64(1)
+
+	addrs := generateTokenAddresses(numAddresses)
+
+	updates := make(map[ccip.Address][]TokenPriceUpdate)
+	for _, addr := range addrs {
+		updates[addr] = generateTokenPriceUpdates(addr, numUpdatesPerAddress)
+	}
+
+	for _, updatesPerAddr := range updates {
+		err := orm.InsertTokenPricesForDestChain(ctx, destSelector, 1, updatesPerAddr)
+		assert.NoError(t, err)
+	}
+
+	time.Sleep(1 * time.Millisecond)
```

Comment:
What happens if you remove these sleeps? Does it affect the test?

Reply:
Removed and the test can still consistently pass